src/ImageStorage/Image/Transformation/WatermarkRand.php

```diff
@@ -0,0 +1,168 @@
+<?php
+namespace ImageStorage\Image\Transformation;
+
+class WatermarkRand implements \ImageStorage\Image\Transformation
+{
+	/**
+	 * @var \ImageStorage\Image\Structure\Image
+	 */
+	private $_imageStruct;
+
+	/**
+	 * @var \ImageStorage\Image\Structure\WatermarkRand|null
+	 */
+	private $_watermarkRand = null;
+
+	/**
+	 * @param \ImageStorage\Image\Structure\WatermarkRand $watermark
+	 *
+	 * @throws \Exception
+	 */
+	public function __construct(\ImageStorage\Image\Structure\WatermarkRand $watermarkRand = null)
+	{
+		if (!($watermarkRand instanceof \ImageStorage\Image\Structure\WatermarkRand))
+		{
+			throw new \Exception('Bad structure!');
+		}
+
+		$this->_watermarkRand = $watermarkRand;
+	}
+
+	/**
+	 * @param \ImageStorage\Image\Structure\Image $imageStruct
+	 *
+	 * @return \ImageStorage\Image\Structure\Image
+	 */
+	public function transform(\ImageStorage\Image\Structure\Image $imageStruct)
+	{
+		$this->_imageStruct = $imageStruct;
```

Comment:
it’s better to pass this structure to the private method… Why exactly do you have a separate private method instead of implementing it here?